blocks/log/json.py

```diff
@@ -0,0 +1,100 @@
+from mimir import Logger
+from mimir.logger import _Logger
+
+from .log import TrainingLogBase
+
+
+class PicklableLogger(_Logger):
+    """A picklable wrapper around mimir logger.
+
+    This class is a picklable version of `:class:mimir.Logger`.
+
+    """
+    def __init__(self, **kwargs):
+        self.logger_kwargs = kwargs
+        self.opened = False
+
+    def open(self):
+        if not self.opened:
+            logger = Logger(**self.logger_kwargs)
+            self.__dict__.update(logger.__dict__)
+            self.load(self.logger_kwargs['filename'])
+            self.opened = True
+
+    def __setstate__(self, state):
+        self.logger_kwargs = state
+        self.opened = False
+        self.open()
+
+    def __getstate__(self):
+        return self.logger_kwargs
+
+
+class JSONLinesLog(TrainingLogBase):
+    """A log stored in gzipped JSON Lines format.
+
+    Each line of the log is a dictionary of a form
+    `{<iteration>: {<record_name>: <recodr_value>...}}`.
+
+    Examples
+    --------
+
+    Analysis of the log can be easily done with
+    `jq <https://stedolan.github.io/jq/>`__
+
+    .. code:: bash
+        gunzip -c log.jsonl.gz | jq '.[].train_error'
+
+        # Or equivalently
+        zcat log.jsonl.gz | jq '.[].train_error'
+
+        # To filter out null entires
+        zcat log.jsnol.gz | jq '.[].train_error | select(.>0)'
+
+    """
+    def __init__(self, filename='log.jsonl.gz', **kwargs):
+        self.status = {}
+        TrainingLogBase.__init__(self)
+        kwargs.setdefault("maxlen", 21)
+        kwargs.setdefault("filename", filename)
```

Comment:
You don' have to do this, Python will complain if a user passes filename as a positional as well as a keyword argument.

I think it would be cleaner to just use `__init__(self, filename='log.jsonl.gz', maxlen=21, formatter=None, **kwargs)` though and then call `PicklableLogger(filename=filename, maxlen=maxlen, formater=formatter, **kwargs)`. Better for default arguments to be in the signature than hidden in the code.